src/mscorlib/shared/System/Collections/Concurrent/CDSCollectionETWBCLProvider.cs

```diff
@@ -0,0 +1,116 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+// =+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+
+//
+// CDSCollectionETWBCLProvider.cs
+//
+// A helper class for firing ETW events related to the Coordination Data Structure 
+// collection types. This provider is used by CDS collections in both mscorlib.dll 
+// and System.dll. The purpose of sharing the provider class is to be able to enable 
+// ETW tracing on all CDS collection with a single ETW provider GUID, and to minimize
+// the number of providers in use.
+//
+// =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
+
+using System.Diagnostics.Tracing;
+
+namespace System.Collections.Concurrent
+{
+    /// <summary>Provides an event source for tracing CDS collection information.</summary>
+    [EventSource(
+        Name = "System.Collections.Concurrent.ConcurrentCollectionsEventSource",
+        Guid = "35167F8E-49B2-4b96-AB86-435B59336B5E"
+        //TODO:Bug455853:Add support for reading localized string in the EventSource il2il transform
+        //,LocalizationResources = "mscorlib"
+        )]
+    internal sealed class CDSCollectionETWBCLProvider : EventSource
+    {
+        /// <summary>
+        /// Defines the singleton instance for the collection ETW provider.
+        /// The collection provider GUID is {35167F8E-49B2-4b96-AB86-435B59336B5E}.
+        /// </summary>
+        public static CDSCollectionETWBCLProvider Log = new CDSCollectionETWBCLProvider();
+        /// <summary>Prevent external instantiation.  All logging should go through the Log instance.</summary>
+        private CDSCollectionETWBCLProvider() { }
```

Comment:
As the ConcurrentQueue is used by the ThreadPool so very commonly initialised in all apps; use the faster start-up base constructor for the `EventSource` (e.g. not via reflection)? re: https://github.com/dotnet/coreclr/issues/15954
```csharp
private CDSCollectionETWBCLProvider() 
: base(new Guid(0x35167F8E, 0x49B2, 0x4b96, 0xAB, 0x86, 0x43, 0x5B, 0x59, 0x33, 0x6B, 0x5E), 
  "System.Collections.Concurrent.ConcurrentCollectionsEventSource") { }
```

/cc @stephentoub